inc/config/http-datasource/http-datasource-interface.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * HttpDatasourceInterface
+ *
+ * @package remote-data-blocks
+ * @since 0.1.0
+ */
+
+namespace RemoteDataBlocks\Config;
+
+defined( 'ABSPATH' ) || exit();
+
+/**
+ * Interface used to define a Remote Data Blocks Datasource for an HTTP API. It
+ * defines the properties of an API that will be shared by queries against that
+ * API.
+ *
+ * Assumptions:
+ * - The API speaks valid JSON, both for request and response bodies.
+ * - The API returns 2XX for successful requests.
+ * - The API returns 3XX for redirects with a maximum of 5 redirects.
+ * - The API returns 4XX or 5XX for unrecoverable errors, in which case the
+ *   response should be ignored.
+ *
+ * If you are a WPVIP customer, datasources are automatically provided by VIP.
+ * Only implement this interface if you have custom datasources not provided by VIP.
+ */
+interface HttpDatasourceInterface {
+
+	/**
+	 * Constructor for the HttpDatasource.
+	 *
+	 * @param array $config The configuration object for this HTTP datasource.
+	 */
+	public function __construct( array $config );
+
+	/**
+	 * Get a human-readable name for this datasource.
```

Comment:
I'm torn about where these comments are most useful, in the interface or the implementation? Where are users more likely to look? Maybe both?